NMDSdevopsServiceAdm/DataEngineering
# Sphinx exploration

# Description

The POC content for our new low level documentation rendered by Sphinx, including all of the steps taken to build it from scratch included on a walkthrough page (which can be later removed or transferred to Confluence).
In the codebase will be several useful functions added deliberately for our use-cases.
There is a change to 2 docstrings in utils for autodoc rendering purposes.

# How to test

- Pull down the branch, then using your vscode terminal (or otherwise).
- Marion commented you might need to update your piplock file first, so try either of the following:
-- Try `pipenv sync --dev` to update your environment dependencies. 
-- OR try `pipenv lock --clear` , then `pipenv build --dev`
- run `sphinx-autobuild docs/source/ docs/build/`
- In your terminal, you should get a dialogue that the server is running on localhost on some port, open a browser and copy and paste that url in
- You should be able to see the Sphinx POC there.

From then onwards, you only need to run the one command `sphinx-autobuild docs/source/ docs/build/` to render the documentation locally.

# Developer checklist
- [ ] Documentation up to date <- Need to update the current README with the sphinx instructions mentioned above.
